GuideForNewDevelopers.md

```diff
@@ -0,0 +1,559 @@
+    - [1.1.8 Install requirements](#118-install-requirements)
+    - [1.1.9 Run the project](#119-run-the-project)
+    - [1.1.9.1 Open terminal and navigate inside the project directory](#1191-open-terminal-and-navigate-inside-the-project-directory)
+    - [1.1.9.2  Activate virtual python environment.](#1192--activate-virtual-python-environment)
+    - [1.1.9.3 Copy .env.example file to .env](#1193-copy-envexample-file-to-env)
+    - [1.1.9.4 Run the following commands to setup the database](#1194-run-the-following-commands-to-setup-the-database)
+    - [1.1.10 Start the server(Finally)](#1110-start-the-serverfinally)
+  - [1.2 Windows](#12-windows)
+    - [1.2.1 Install Python3](#121-install-python3)
+    - [1.2.2 Install git Bash](#122-install-git-bash)
+    - [1.2.3 Install VSCode](#123-install-vscode)
+    - [1.2.4 Clone the Project Locally](#124-clone-the-project-locally)
+    - [1.2.5 Install and Create Virtual Environment](#125-install-and-create-virtual-environment)
+    - [1.2.6 Install requirements](#126-install-requirements)
+    - [1.2.7 Run the project](#127-run-the-project)
+    - [1.2.7.1 Open CMD and navigate inside the project directory](#1271-open-cmd-and-navigate-inside-the-project-directory)
+    - [1.2.8.2  Activate virtual python environment.](#1282--activate-virtual-python-environment)
+    - [1.2.8.3 Copy .env.example file to .env](#1283-copy-envexample-file-to-env)
+    - [1.2.8.4 Run the following commands to setup the database](#1284-run-the-following-commands-to-setup-the-database)
+    - [1.2.10 Start the server(Finally)](#1210-start-the-serverfinally)
+  - [1.3 Linux](#13-linux)
+    - [1.3.1 Install Python3](#131-install-python3)
+    - [1.3.2 Install git](#132-install-git)
+    - [1.3.3 Install sqlite3](#133-install-sqlite3)
+    - [1.3.4 Install VSCode](#134-install-vscode)
+    - [1.3.5 Clone the Project Locally](#135-clone-the-project-locally)
+    - [1.3.6 Install and Create Virtual Environment](#136-install-and-create-virtual-environment)
+    - [1.3.7 Install requirements](#137-install-requirements)
+    - [1.3.8 Run the project](#138-run-the-project)
+    - [1.3.8.1 Open terminal and navigate inside the project directory](#1381-open-terminal-and-navigate-inside-the-project-directory)
+    - [1.3.8.2  Activate virtual python environment.](#1382--activate-virtual-python-environment)
+    - [1.3.8.3 Copy .env.example file to .env](#1383-copy-envexample-file-to-env)
+    - [1.3.8.4 Run the following commands to setup the database](#1384-run-the-following-commands-to-setup-the-database)
+    - [1.3.10 Start the server(Finally)](#1310-start-the-serverfinally)
+- [Working on a new features](#working-on-a-new-features)
+- [Troubleshooting[WIP]](#troubleshootingwip)
+  - [MAC](#mac)
+    - [Homebrew not found](#homebrew-not-found)
+  - [Windows](#windows)
+  - [Linux](#linux)
+
+
+
+
+
+# 1. Setting up your system for the first time
```

Comment:
I think a high level synopsis here would be great. Something like:


Development requires the following software:

- a package management system (e.g. homebrew)
- Python
- git
- sqlite3
- an integrated development environment (e.g. VSCode)